### DISCUSSION TITLE
mitmweb via Docker and addon: 'no such script'

### DISCUSSION THREAD

**Discussion by malwareinfosec** [mitmproxy/mitmproxy#7235] [Q&A] [answered]
Hello,

I'm running mitmweb via Docker using the following command:

`docker run --rm -it -p 8080:8080 -p 127.0.0.1:8081:8081 mitmproxy/mitmproxy mitmweb --web-host 0.0.0.0`

It works fine. Now I want to use an addon, so I append the -s parameter with the name of the addon:

`docker run --rm -it -p 8080:8080 -p 127.0.0.1:8081:8081 mitmproxy/mitmproxy mitmweb --web-host 0.0.0.0 -s myscript.py`

However, I am getting the following error:

`/usr/local/bin/mitmweb: No such script: myscript.py`

I don't understand where the script should be located (I've tried to move it to various locations) including the path where I run the command from.

Maybe this is more of a Docker question, apologies if so as I am new to it.

**Comment by Prinzhorn** [comment, marked as answer]
> Maybe this is more of a Docker question

Yes, very much so.

There are multiple options. You can mount a local folder into the container (so you can update the script without changing the container) or you can use `docker cp` to copy the file into the container. But since you're using `--rm` that won't really work.

See https://docs.docker.com/engine/storage/bind-mounts/ for option 1 e.g. `  -v /path/to/your/scripts/folders:/scripts -s /scripts/myscriptpy`

**Reply by malwareinfosec** [reply]
Thanks!

Here's what I did. I created a Dockerfile with the following:
COPY fiddleitm.py /fiddleitm/
CMD ["mitmweb", "-s", "/fiddleitm/fiddleitm.py"]

and then I run:
``docker run --rm -it -v /home/user/fiddleitm:/home/mitmproxy/.mitmproxy -p 8080:8080 fiddleitm``

**Reply by Prinzhorn** [reply]
That's the best option when your script is static. The two options I provided allow working with scripts outside of the container when it is already running.
